src/response.rs

```diff
@@ -439,28 +439,44 @@ impl Response {
 fn parse_status_line(line: &str) -> Result<(ResponseStatusIndex, u16), Error> {
     //
 
-    let mut split = line.splitn(3, ' ');
+    let err = Err(BadStatus.new());
```

Comment:
Could we maybe move this inline and give some relevant error messages? I.e.

`BadStatus.msg("Status line is not ASCII")`, `BadStatus.msg("Version does not start HTTP/")`, etc.